src/updates.js

```diff
@@ -210,6 +224,7 @@ class Updates {
     for (const release of releases) {
       if (
         !semver.valid(release.tag_name) ||
+        semver.lt(release.tag_name, version) ||
```

Comment:
This change in behavior seems unrelated? Can you explain this change and why it's needed in this PR

Reply:
Thanks for the review @MarshallOfSound - apologies for not calling out this change.

This was important to ensure that assets from previous releases weren't being matched.

For example, without this change `http://localhost:62178/gitify-app/gitify/darwin-x64/5.14.0` and `http://localhost:62178/gitify-app/gitify/darwin-arm64/5.14.0` were both finding the "latest" assets from a really old release https://github.com/gitify-app/gitify/releases/tag/v4.2.1 - which was the latest time we published non-universal artifacts. This then causes the following empty response
https://github.com/electron/update.electronjs.org/blob/3638132373d5e5c7e596261832ff543caecb9f36/src/updates.js#L110-L113 

Adding in the above check ensures that only newer versions are considered eligible for the `latest` assets object

Reply:
This isn't an issue with your other changes in this PR right, i.e. you don't need this now that you've added `universal` fallback?

The regex changes seem safe at a glance, but changing this `lte` from it's original defense location would require more thinking. If you remove this `lt` change reviewing this PR will be much easier and will be more likely it lands.

Reply:
Actually, this change is required as part of this PR for it to work.

Without it, `latest` is not null, therefore it never falls back to checking if the universal release.

Reply:
Hm, can we modify the fallback logic from: "get x64/arm64, if not exists get universal" or just consider `unviersal` builds `arm64/x64` builds by default if no arm64/x64 specific build is present. i.e. it's a fallback within get asset, rather than a separate call?

Reply:
i.e. in here https://github.com/electron/update.electronjs.org/blob/3638132373d5e5c7e596261832ff543caecb9f36/src/updates.js#L175

Just do effective `if (!latest[platform] && platform === A_DARWIN_PLATFORM && latest[universal]) return latest[universal]`

Reply:
Removed the `lt` check and changes where the latestUniversal replacement occurs